src/main/host/process.rs

```diff
@@ -510,6 +528,71 @@ impl RunnableProcess {
         host.schedule_task_with_delay(task, SimulationTime::ZERO);
     }
 
+    pub fn new_forked_process(
+        &self,
+        host: &Host,
+        new_tgl: RootedRc<RootedRefCell<Thread>>,
+    ) -> RootedRc<RootedRefCell<Process>> {
+        let new_tgl_tid;
+        let native_pid;
+        {
+            let new_tgl = new_tgl.borrow(host.root());
+            new_tgl_tid = new_tgl.id();
+            native_pid = new_tgl.native_pid();
+        }
+        let pid = ProcessId::from_thread_group_leader_tid(new_tgl_tid);
+        let plugin_name = self.common.plugin_name.clone();
+        let name = make_name(host, plugin_name.to_str().unwrap(), pid);
+
+        let common = Common {
+            id: pid,
+            host_id: host.id(),
+            name,
+            plugin_name,
+            working_dir: self.common.working_dir.clone(),
+        };
+
+        // The child will log to the same strace log file. Entries contain thread IDs,
+        // though it might be tricky to map those back to processes.
+        let strace_logging = self
+            .strace_logging
+            .as_ref()
+            .map(|strace| strace.try_clone().unwrap());
+
+        // `fork(2)`:
+        //  > The child does not inherit timers from its parent
+        //  > (setitimer(2), alarm(2), timer_create(2)).
+        let itimer_real = RefCell::new(Timer::new(move |host| itimer_real_expiration(host, pid)));
+
+        let threads = RefCell::new(BTreeMap::from([(new_tgl_tid, new_tgl)]));
+
+        let shim_shared_mem = ProcessShmem::new(
+            &host.shim_shmem_lock_borrow().unwrap().root,
+            host.shim_shmem().serialize(),
+            host.id(),
+            strace_logging.as_ref().map(|x| x.file.borrow().as_raw_fd()),
+        );
+        let shim_shared_mem_block = shadow_shmem::allocator::shmalloc(shim_shared_mem);
+
+        let runnable_process = RunnableProcess {
+            common,
+            expected_final_state: None,
```

Comment:
Maybe update the documentation to explain that `expected_final_state` only applies to the process in the config, and not any forked processes? And explain how the final state of forked processes is handled? (Is it ignored?)